sdk/appconfig/azappconfig/client.go

```diff
@@ -403,70 +401,29 @@ func fromGeneratedGetRevisionsPage(g generated.AzureAppConfigurationClientGetRev
 	}
 }
 
-// ListRevisionsPager is a Pager for revision list operations.
-type ListRevisionsPager struct {
-	options   *generated.AzureAppConfigurationClientGetRevisionsOptions
-	genClient *generated.AzureAppConfigurationClient
-	nextLink  *string
-}
-
-// More returns true if there are more pages to return
-func (l *ListRevisionsPager) More() bool {
-	if !reflect.ValueOf(l.nextLink).IsZero() {
-		if l.nextLink == nil || len(*l.nextLink) == 0 {
-			return false
-		}
-	}
-	return true
-}
-
-// NextPage returns the current page of results
-func (l *ListRevisionsPager) NextPage(ctx context.Context) (ListRevisionsPage, error) {
-	var resp *http.Response
-	var err error
-	if l.nextLink == nil {
-		req, err := l.genClient.GetRevisionsCreateRequest(
-			ctx,
-			l.options,
-		)
-		if err != nil {
-			return ListRevisionsPage{}, err
-		}
-		resp, err = l.genClient.Pl.Do(req)
-		if err != nil {
-			return ListRevisionsPage{}, err
-		}
-	} else {
-		req, err := runtime.NewRequest(ctx, http.MethodGet, *l.nextLink)
-		if err != nil {
-			return ListRevisionsPage{}, err
-		}
-		resp, err = l.genClient.Pl.Do(req)
-		if err != nil {
-			return ListRevisionsPage{}, err
-		}
-	}
-	if err != nil {
-		return ListRevisionsPage{}, err
-	}
-	result, err := l.genClient.GetRevisionsHandleResponse(resp)
-	if err != nil {
-		return ListRevisionsPage{}, err
-	}
-	if result.NextLink == nil {
-		// Set it to the zero value
-		result.NextLink = to.StringPtr("")
-	}
-	l.nextLink = result.NextLink
-	return fromGeneratedGetRevisionsPage(result), nil
-}
-
 // ListRevisionsOptions contains the optional parameters for the ListRevisions method.
 type ListRevisionsOptions struct {
 	// placeholder for future options
 }
 
-// ListRevisions retrieves the revisions of one or more configuration setting entities that match the specified setting selector.
-func (c *Client) ListRevisions(selector SettingSelector, options *ListRevisionsOptions) ListRevisionsPager {
-	return ListRevisionsPager{options: selector.toGenerated(), genClient: c.appConfigClient, nextLink: nil}
+// NewListRevisionsPager creates a pager that retrieves the revisions of one or more
+// configuration setting entities that match the specified setting selector.
+func (c *Client) NewListRevisionsPager(selector SettingSelector, options *ListRevisionsOptions) *runtime.Pager[ListRevisionsPage] {
+	pagerInternal := c.appConfigClient.NewGetRevisionsPager(selector.toGenerated())
+	return runtime.NewPager(runtime.PageProcessor[ListRevisionsPage]{
+		More: func(ListRevisionsPage) bool {
+			return pagerInternal.More()
+		},
+		Fetcher: func(ctx context.Context, cur *ListRevisionsPage) (ListRevisionsPage, error) {
+			page, err := pagerInternal.NextPage(ctx)
+			if err != nil {
+				return ListRevisionsPage{}, err
+			}
+			if page.NextLink == nil {
+				// Set it to the zero value
+				page.NextLink = to.Ptr("")
```

Comment:
@antkmsft I kept this from the original implementation, but is it really necessary?

Reply:
Will leave for now.